clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -894,55 +944,36 @@ raw2trace_t::process_syscall_pt(raw2trace_thread_data_t *tdata, uint64_t syscall
         tdata->pt_metadata_processed = true;
     }
 
-    if (tdata->pre_read_pt_entries.empty() && tdata->kthread_file->eof()) {
+    if (pt_data == nullptr) {
+        if (!tdata->error.empty())
+            return false;
         VPRINT(1, "Finished decoding all PT data for thread %d\n", tdata->tid);
         return true;
     }
+    uint64_t syscall_args_num =
+        pt_data->header[PDB_HEADER_NUM_ARGS_IDX].syscall_args_num.args_num;
+    uint64_t pt_data_size =
+        pt_data->header[PDB_HEADER_DATA_BOUNDARY_IDX].pt_data_boundary.data_size -
+        PT_DATA_PDB_HEADER_SIZE - syscall_args_num * sizeof(uint64_t);
 
-    if (tdata->pre_read_pt_entries.empty()) {
```

Comment:
Confused about where this line went in the refactoring?

Reply:
It was removed.

We read the header and the data together since they always occur together.  I don't see any path where we read just the header only. Further, since pre_read_pt_entries is used locally only, it doesn't need to be in tdata.

Reply:
Even if we make a change to write the header and data in separate writes, they will still be read back-to-back in code here.